packages/editor/src/components/hierarchy/HierarchyPanelContainer.tsx

```diff
@@ -918,231 +130,147 @@ export default function HierarchyPanel() {
-      if (e.key === 'ArrowDown') {
-        e.preventDefault()
+    (e: KeyboardEvent, node: HeirarchyTreeNodeType) => {
+      const nodeIndex = nodes.indexOf(node)
 
-        // initializing nodeIndex using nodes array
-        const nodeIndex = nodes.indexOf(node)
+      switch (e.key) {
+        case 'ArrowDown':
+          e.preventDefault()
 
-        // initializing nextNode using nodes array
-        const nextNode = nodeIndex !== -1 && nodes[nodeIndex + 1]
+          const nextNode = nodeIndex !== -1 && nodes[nodeIndex + 1]
+          if (!nextNode) return
 
-        // check if nextNode is not Empty
-        if (nextNode) {
           if (e.shiftKey) {
-            CommandManager.instance.executeCommandWithHistory(EditorCommands.ADD_TO_SELECTION, nextNode.object)
+            CommandManager.instance.executeCommandWithHistory(EditorCommands.ADD_TO_SELECTION, nextNode.entityNode)
           }
-          // initializing nextNodeEl using nextNode element id
-          const nextNodeEl = document.getElementById(getNodeElId(nextNode))
 
-          //check if nextNodeEl is not empty
-          if (nextNodeEl) {
-            nextNodeEl.focus()
-          }
-        }
-
-        //check if pressed key equals to ArrowUp
-      } else if (e.key === 'ArrowUp') {
-        e.preventDefault()
+          const nextNodeEl = document.getElementById(getNodeElId(nextNode))
+          if (nextNodeEl) nextNodeEl.focus()
+          break
 
-        // initializing nodeIndex  using nodes array
-        const nodeIndex = nodes.indexOf(node)
+        case 'ArrowUp':
+          e.preventDefault()
 
-        // initializing prevNode using nodes array current nodeIndex -1
-        const prevNode = nodeIndex !== -1 && nodes[nodeIndex - 1]
+          const prevNode = nodeIndex !== -1 && nodes[nodeIndex - 1]
+          if (!prevNode) return
 
-        // if prevNode is not empty
-        if (prevNode) {
           if (e.shiftKey) {
-            CommandManager.instance.executeCommandWithHistory(EditorCommands.ADD_TO_SELECTION, prevNode.object)
+            CommandManager.instance.executeCommandWithHistory(EditorCommands.ADD_TO_SELECTION, prevNode.entityNode)
           }
 
-          // initializing prevNodeEl using prevNode Id
           const prevNodeEl = document.getElementById(getNodeElId(prevNode))
+          if (prevNodeEl) prevNodeEl.focus()
+          break
 
-          //check if prevNodeEl is not empty
-          if (prevNodeEl) {
-            prevNodeEl.focus()
-          }
-        }
-        // check if pressed key equals to left arrow and node object children node length is greator then 0
-      } else if (e.key === 'ArrowLeft' && node.object.children.filter((o) => o.isNode).length > 0) {
-        if (e.shiftKey) {
-          collapseChildren(node)
-        } else {
-          collapseNode(node)
-        }
+        case 'ArrowLeft':
+          if (!node.entityNode.children || node.entityNode.children.length === 0) return
```

Comment:
`node.entityNode?.children?.length === 0` will return the same response

Reply:
When the `children` will be `undefined` then condition will become `false` instead of `true`